### PR TITLE
Fixed Arbitrary Code Injection

### PR DESCRIPTION
**Pull request by v1dhun**
Added an if condition to verify command input using regex matching.

https://github.com/V1dhun/listening-processes/blob/bf07a70ff9fcbefe478c36eaae1976639b489949/index.js#L5

IMG

### CODE REVIEW

**Comment by V1dhun** [comment]
Updated PR #3

File: index.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
 const execSync = require('child_process').execSync
 
 function getProcesses (command) {
-  const lsofOutput = execSync(`lsof -i TCP -P -n | grep '${command}\\s.*:[0-9]* (LISTEN)' | cat`, {encoding: 'utf-8'})
-    .toString()
-    .split('\n')
   const processesObject = {}
+  const regex = /[^A-Za-z0-9 .\-_]+/g
```
Comment (toufik-airane) [inline index.js:5]: The regexp doesn't seem to work with a potentially legit input.
The code line `"service".match(/[^A-Za-z0-9 .\-_]+/g)` return `null`.

**Review by toufik-airane — COMMENTED** [review]
Dear @V1dhun,

Thank you for your contribution, however, I couldn't successfully pass the regular expression with potential legit input.
Would you mind to provide further user acceptant testing?

Best regards.

File: index.js (hunk 1 above)
Comment (v1dhun) [reply index.js:5]: Hi @toufik-airane,
```javascript
if(command.match(regex)){
    return processesObject
  }
```
The if condition is supposed to catch only illegal characters. Completely legit inputs would go with else condition. So it's works Perfectly.

**Review by toufik-airane — APPROVED** [review]
LGTM.

**Review by JamieSlome — APPROVED** [review]
Thanks for your contribution - good job! 🍰

**Comment by huntr-helper** [comment]
Thank you for submitting your pull request. On this occasion, one of your fellow huntrs has been awarded the bounty. We appreciate your effort and look forward to reviewing more of your fixes in the future!

**Pull request closed without merging**